### PR TITLE
MBS-11774 Replace utilization metric by living,working,idle

### PR DESCRIPTION
**Pull request by sergioboi**
it will help to count idle time and utilization for all tasks

### CODE REVIEW

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/metrics/TestRunnerMetricsListener.kt [hunk 1]
```diff
@@ -133,11 +133,9 @@ internal class TestRunnerMetricsListener(
             )
 
             sendTotalTime(provider.totalTime())
-
-            provider.medianDeviceUtilization().fold(
-                { sendMedianDeviceUtilization(it) },
-                { logger.warn("Not sending median device relative wasted time, no data") }
-            )
+            sendDevicesLiving(provider.devicesLiving())
```
Comment (sergioboi) [inline subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/metrics/TestRunnerMetricsListener.kt:136]: So, why did I decide to send 3 different indicators comparing to `device-utilization`?

1. Indicators are absolute. So, if we sum series for all tasks we will get `device-utilization` for all tasks
2. We could count the `idle-time-budget`. The amount of time when our devices do nothing across all tasks and devices

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/metrics/model/DeviceWorkerState.kt [hunk 2]
```diff
@@ -79,13 +77,15 @@ internal sealed class DeviceWorkerState {
         override val key: DeviceKey,
     ) : DeviceWorkerState() {
 
-        private val totalTime: Duration = Duration.between(created, finished)
+        val livingTime: Duration = Duration.between(created, finished)
```
Comment (sergioboi) [inline subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/metrics/model/DeviceWorkerState.kt:80]: For me, living, working, and idle simpler than total and effective

File: subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/metrics/model/TestExecutionState.kt [hunk 3]
```diff
@@ -51,7 +51,7 @@ internal sealed class TestExecutionState {
         val completed: Instant,
         override val test: TestKey
     ) : TestExecutionState() {
-        val effectiveWorkTime: Duration = Duration.between(intentionReceived, completed)
+        val processingTime: Duration = Duration.between(intentionReceived, completed)
```
Comment (sergioboi) [inline subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/metrics/model/TestExecutionState.kt:54]: effective is an assumption

**Review by dsvoronin — APPROVED** [review]

File: docs/content/test_runner/Metrics.md [hunk 4]
```diff
@@ -6,25 +6,48 @@ Metrics available at:
...
-    subgraph ef1[effective part 1]
-    id2 --> id3[Test started] --> id4[Test finished]
+    subgraph ef1[Living]
+    id1[Device started] --> id2[Device finished]
+    end 
+```
+
+### `devices.working`
+
+Sum of all time when device executing a test across all devices
```
Comment (eugene-krivobokov) [inline docs/content/test_runner/Metrics.md]:
```suggestion
Sum of all time when device executing tests across all devices
```

**Pull request merged**